contract/src/main/java/io/neow3j/contract/TransactionBuilder.java

```diff
@@ -391,56 +385,42 @@ private long getSystemFeeForScript() throws IOException {
-     * signers added to the transaction.
-     */
     private long calcNetworkFee() throws IOException {
-        Transaction tx = new Transaction(neow, version, nonce, validUntilBlock, signers, 0,
-                0, attributes, script, new ArrayList<>());
-        getSignerAccounts().forEach(s -> {
-            tx.addWitness(new Witness(new byte[]{}, s.getVerificationScript().getScript()));
-        });
-        NeoCalculateNetworkFee networkFeeResult = neow
-                .calculateNetworkFee(Numeric.toHexStringNoPrefix(tx.toArray())).send();
-        BigInteger networkFee = networkFeeResult.getNetworkFee().getNetworkFee();
-        return networkFee.longValue();
-    }
-
-    /**
-     * Gets the signer accounts held in the wallet.
-     *
-     * @return a list containing the signer accounts
-     */
-    private List<Account> getSignerAccounts() {
-        List<Account> sigAccounts = new ArrayList<>();
-        signers.forEach(signer -> {
-            if (wallet.holdsAccount(signer.getScriptHash())) {
-                sigAccounts.add(wallet.getAccount(signer.getScriptHash()));
+        Transaction tx = new Transaction(neow, version, nonce, validUntilBlock, signers, 0, 0,
+                attributes, script, new ArrayList<>());
+        // For each signer that is available in the wallet and has a verification script, we add
+        // a witness to a temporary transaction object that is serialized and sent to the
+        // `getnetworkfee` RPC method. Signers that are contract's do not need a verification
+        // script. Instead, their `verify` method will be consulted by the neo-node. We simply
+        // add an empty witness for such signers, resp. signers that don't have a verification
+        // script in the wallet.
+        boolean hasAtLeastOneSigningAccount = false;
+        for (Signer signer : signers) {
+            Account a = wallet.getAccount(signer.getScriptHash());
+            if (a != null && a.getVerificationScript() != null) {
+                tx.addWitness(new Witness(new byte[]{}, a.getVerificationScript().getScript()));
+                hasAtLeastOneSigningAccount = true;
             } else {
-                throw new TransactionConfigurationException(
-                        "Cannot find account with script hash '" +
-                        signer.getScriptHash().toString() + "' in wallet set on this transaction " +
-                        "builder.");
+                tx.addWitness(new Witness());
             }
-        });
-        return sigAccounts;
+        }
+        if (!hasAtLeastOneSigningAccount) {
+            throw new TransactionConfigurationException("No signers were set for which an account" +
+                    " with verification script exists in the wallet.");
+        }
+        String txHex = Numeric.toHexStringNoPrefix(tx.toArray());
+        return neow.calculateNetworkFee(txHex).send().getNetworkFee().getNetworkFee().longValue();
     }
```

Comment:
Important change to review.

Reply:
Alright. I think this is the logic I was expecting! Cool. 💪